inkcpp/include/globals.h

```diff
@@ -4,15 +4,148 @@
 
 namespace ink::runtime
 {
+	class globals_interface;
+	namespace internal { class globals_impl;}
+	/**
+	 * @brief wrapper for string in global store.
+	 * @tparam WRITE if writing enable for this type
+	 */
+	template<bool WRITE = false>
+	class global_string {
+	public:
+		operator bool() { return _string; }
+		auto operator<=>(const global_string& other) const;
+		char operator[](size_t i) const { return _string[i]; }; // TODO: boundary check?
+		size_t size() const { return _size; };
+		const char* data() const { return _string; }
+
+	protected:
+		friend class ink::runtime::internal::globals_impl;
+		global_string(const internal::globals_impl& globals, const char* string);
+
+		const internal::globals_impl& _globals;
```

Comment:
This should actually use the shared_ptr `globals` defined in types.h. Remember that it's possible for the globals object to be destroyed while one of these global_strings is in scope (imagine the user reads the value of a variable, stores that somewhere, then lets the ink story and runners die). The shared pointer can guard you against that. Just make sure you check it for validity and maybe throw an exception if they try to do a write when it's not valid.